incubator/hnc/pkg/object/object.go

```diff
@@ -26,7 +26,7 @@ func Canonical(inst *unstructured.Unstructured) *unstructured.Unstructured {
 	// Non-HNC annotations:
 	newAnnots := map[string]string{}
 	for k, v := range inst.GetAnnotations() {
-		if !strings.HasPrefix(k, api.MetaGroup) {
+		if !strings.Contains(k, api.MetaGroup) {
```

Comment:
I don't think this is the right logic. While it's unlikely, `hnc.x-k8s.io.foo/bar` is actually fine, and shouldn't be stripped out. Maybe add this as a test case too?

Reply:
sg.